### PR TITLE
backing up smaug opf-observatorium into git

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
After looking into the [loki is not serviceable issue](https://github.com/operate-first/support/issues/604), it seemed like a good idea to backup our openshift manifests from opf-observatorium in the Smaug cluster into git.

/cc @harshad16 
/cc @HumairAK

### CODE REVIEW

**Comment by Gregory-Pereira** [comment]
@humair, do the deployments go in base and get patched in the Smaug overlay, or is putting the deployment in overlays sufficient?

**Comment by Gregory-Pereira** [comment]
Ready for review -->
/unhold
Part of review is making sure all external secrets in `opf-observatorium` of Smaug cluster have been replicated correctly in vault. Extra attention should be given to this, as I was uncertain which secret values should be `base64` encoded and which should not be.

File: observatorium/base/instance/loki/common/observatorium-loki-configmap.yaml [hunk 1]
```diff
@@ -129,3 +129,4 @@ data:
 kind: ConfigMap
 metadata:
   name: observatorium-loki
+  namespace: opf-observatorium
```
Comment (harshad16) [inline observatorium/base/instance/loki/common/observatorium-loki-configmap.yaml:132]: we should keep the hard value of namespace defined in here
either we would rely on kustomization.yaml or argocd application.

please drop this.

File: observatorium/overlays/moc/smaug/loki/configmaps/observatorium-api-rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: observatorium-api
+  namespace: opf-observatorium
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/loki/configmaps/observatorium-api-rbac.yaml:5]: drop the namespace here aswell.

File: observatorium/overlays/moc/smaug/loki/configmaps/observatorium-loki-configmap_patch.yaml [hunk 3]
```diff
@@ -120,12 +121,7 @@ data:
...
-  overrides.yaml: '{}'
 kind: ConfigMap
 metadata:
-  labels:
-    app.kubernetes.io/instance: observatorium
-    app.kubernetes.io/name: loki
-    app.kubernetes.io/part-of: observatorium
-    app.kubernetes.io/version: 2.3.0
   name: observatorium-loki
+  namespace: opf-observatorium
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/loki/configmaps/observatorium-loki-configmap_patch.yaml:127]: drop the namespace here as well.

File: observatorium/overlays/moc/smaug/loki/deployments/loki-query-frontend-oauth-proxy.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: loki-query-frontend-oauth-proxy
+  namespace: opf-observatorium
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/loki/deployments/loki-query-frontend-oauth-proxy.yaml:5]: drop the namespace here as well.

File: observatorium/overlays/moc/smaug/loki/deployments/observatorium-api.yaml [hunk 5]
```diff
@@ -0,0 +1,102 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: observatorium-api
+  namespace: opf-observatorium
+  labels:
+    app.kubernetes.io/component: api
+    app.kubernetes.io/instance: observatorium
+    app.kubernetes.io/name: observatorium-api
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/loki/deployments/observatorium-api.yaml:9]: include these labels directly in kustomization.yaml and drop them from here.

File: observatorium/overlays/moc/smaug/loki/deployments/observatorium-api.yaml [hunk 6]
```diff
@@ -0,0 +1,102 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: observatorium-api
+  namespace: opf-observatorium
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/loki/deployments/observatorium-api.yaml:5]: drop the namespace here as well.

File: observatorium/overlays/moc/smaug/loki/deployments/observatorium-api.yaml [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  labels:
+    app.kubernetes.io/component: api
+    app.kubernetes.io/instance: observatorium
+    app.kubernetes.io/name: observatorium-api
+    app.kubernetes.io/part-of: observatorium
+    app.kubernetes.io/version: master-2021-03-26-v0.1.1-200-gea0242a
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/loki/deployments/observatorium-api.yaml:11]: remove these labeles

File: observatorium/overlays/moc/smaug/loki/deployments/observatorium-loki-distributor.yaml [hunk 8]
```diff
@@ -0,0 +1,146 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: observatorium-loki-distributor
+  namespace: opf-observatorium
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/loki/deployments/observatorium-loki-distributor.yaml:5]: drop the namespace here as well.

File: observatorium/overlays/moc/smaug/loki/obcs/opf-observatorium-metrics.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
+apiVersion: objectbucket.io/v1alpha1
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/loki/obcs/opf-observatorium-metrics.yaml:1]: already seems to be exisiting:
https://github.com/operate-first/apps/blob/3c05ac9fb559f83e6178d47ab16a4b47aa759769/observatorium/overlays/moc/smaug/thanos/obcs/opf-observatorium-metrics.yaml

**Review by harshad16 — CHANGES_REQUESTED** [review]
lgtm for the first iteration.
please remove the namespace reference from individual manifests.

File: observatorium/base/instance/loki/common/observatorium-loki-configmap.yaml (hunk 1 above)
Comment (Gregory-Pereira) [reply observatorium/base/instance/loki/common/observatorium-loki-configmap.yaml]: I can drop this however, this means that we have to simply deploy the `configmap_patch` from the Smaug overlay as part of the kustomization, and cannot patch it. I ran into issues between this file  and the `[observatorium/overlays/moc/smaug/loki/configmaps/observatorium-loki-configmap_patch.yaml](https://github.com/operate-first/apps/pull/2253/files/a375267949230c4d7c6803ede36f1db8d401e255#diff-4d5228adcc74d5cf6bfb0c7191336f0ea69d6bb22d7e3db9c37f95e9d6a3022d) which I resolved by adding the namespace to both to make them unique resources which would allow me to build them in the observatorium/overlays/moc/smaug/loki/configmaps/kustomization.yaml with a `patchesStrategicMerge` directive.

File: observatorium/overlays/moc/smaug/loki/configmaps/observatorium-loki-configmap_patch.yaml (hunk 3 above)
Comment (Gregory-Pereira) [reply observatorium/overlays/moc/smaug/loki/configmaps/observatorium-loki-configmap_patch.yaml:127]: Leaving this unresolved for the same issue as above, but I think the above issue should be solved by the current state of the PR.

Comment (Gregory-Pereira) [reply observatorium/overlays/moc/smaug/loki/configmaps/observatorium-loki-configmap_patch.yaml:126]: Yeah getting the same error as before... here are the logs of the `kustomize-build` check, reverting this change for now: ```
accumulating resources: accumulation err='accumulating resources from 'loki': '/home/prow/go/src/github.com/operate-first/apps/observatorium/overlays/moc/smaug/loki' must resolve to a file': recursed accumulation of path '/home/prow/go/src/github.com/operate-first/apps/observatorium/overlays/moc/smaug/loki': no matches for IdId ~G_v1_ConfigMap|opf-observatorium|observatorium-loki; failed to find unique target for patch ~G_v1_ConfigMap|observatorium-loki
Aborting due to errexit on line [4](https://prow.operate-first.cloud/view/s3/ci-prow/prow-logs/pr-logs/pull/operate-first_apps/2253/kustomize-build/1564763177680375808#1:build-log.txt%3A4)3. Exit code: 1
```

File: observatorium/overlays/moc/smaug/loki/deployments/observatorium-loki-distributor.yaml [hunk 10]
```diff
@@ -0,0 +1,139 @@
+kind: Deployment
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/loki/deployments/observatorium-loki-distributor.yaml:1]: seems like this deployment exists
https://github.com/operate-first/apps/blob/master/observatorium/base/instance/loki/distributor/deployment.yaml

File: observatorium/overlays/moc/smaug/loki/deployments/observatorium-loki-query-frontend.yaml [hunk 11]
```diff
@@ -0,0 +1,124 @@
+kind: Deployment
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/loki/deployments/observatorium-loki-query-frontend.yaml:1]: seems like this already exists on base path
https://github.com/operate-first/apps/blob/master/observatorium/base/instance/loki/query-frontend/deployment.yaml

File: observatorium/overlays/moc/smaug/thanos/deployments/observatorium-thanos-query-frontend.yaml [hunk 12]
```diff
@@ -0,0 +1,104 @@
+kind: Deployment
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/thanos/deployments/observatorium-thanos-query-frontend.yaml:1]: seems like this already exists
https://github.com/operate-first/apps/blob/master/observatorium/base/instance/thanos/query-frontend/deployment.yaml

File: observatorium/overlays/moc/smaug/thanos/deployments/observatorium-thanos-query.yaml [hunk 13]
```diff
@@ -0,0 +1,99 @@
+kind: Deployment
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/thanos/deployments/observatorium-thanos-query.yaml:1]: this already exists
https://github.com/operate-first/apps/blob/master/observatorium/base/instance/thanos/query/deployment.yaml

File: observatorium/overlays/moc/smaug/thanos/deployments/observatorium-thanos-receive-controller.yaml [hunk 14]
```diff
@@ -0,0 +1,65 @@
+kind: Deployment
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/thanos/deployments/observatorium-thanos-receive-controller.yaml:1]: this already exists
https://github.com/operate-first/apps/blob/master/observatorium/base/instance/thanos/receive-controller/deployment.yaml

File: observatorium/overlays/moc/smaug/thanos/obcs/opf-observatorium-metrics.yaml [hunk 15]
```diff
@@ -1,7 +0,0 @@
-apiVersion: objectbucket.io/v1alpha1
```
Comment (harshad16) [inline observatorium/overlays/moc/smaug/thanos/obcs/opf-observatorium-metrics.yaml:1]: why is this deleted?

**Review by harshad16 — CHANGES_REQUESTED** [review]
Please take a closer look at the base path of observatorium https://github.com/operate-first/apps/tree/master/observatorium/base
it seems to be having most of this manifests.

maybe restructure them.
